runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java

```diff
@@ -0,0 +1,220 @@
+import net.runelite.client.game.SpriteManager;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import javax.inject.Inject;
+import net.runelite.client.game.SkillIconManager;
+
+@Slf4j
+public class HpAndStatusOverlay extends Overlay
+{
+	private Client c;
+	private HpAndPrayerConfig config;
+	private SkillIconManager sim;
+	private SpriteManager sm;
+	private FontManager fm;
+
+	final private int HEIGHT = 250;
+	final private int WIDTH = 20;
+	final private int PADDING = 1;
+
+	// Static colors
+	final Color DEPLETED = new Color(25, 25, 25, 150);
+	final Color PRAYER_LEFT = new Color(50, 200, 200, 150);
+	final Color BACKGROUND = new Color(0, 0 , 0, 150);
+	// Dynamic colors
+	Color HEALTH_LEFT = new Color(200, 35, 0, 1);
+	Color POISONED_HEALTH_LEFT = new Color(0, 145, 0, 150);
+	Color VENOMED_HEALTH_LEFT = new Color(0, 65, 0, 150);
+	Color COUNTER_COLOR = new Color(255, 255,  255, 150);
+
+	@Inject
+	public HpAndStatusOverlay(Client c, SpriteManager sm, HpAndPrayerConfig config, SkillIconManager sim, FontManager fm)
+	{
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.c = c;
+		this.sm = sm;
+		this.config = config;
+		this.sim = sim;
+		this.fm = fm;
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		renderInventoryBars(g);
+		return null;
+	}
+
+	private void renderInventoryBars(Graphics2D g)
+	{
+		//Variables
+		int max_health = c.getRealSkillLevel(Skill.HITPOINTS);
+		int max_prayer = c.getRealSkillLevel(Skill.PRAYER);
+		int current_health = c.getBoostedSkillLevel(Skill.HITPOINTS);
+		int current_prayer = c.getBoostedSkillLevel(Skill.PRAYER);
+		int poison_state = c.getVar(VarPlayer.IS_POISONED);
+		int counter_health = c.getBoostedSkillLevel(Skill.HITPOINTS);
+		int counter_prayer = c.getBoostedSkillLevel(Skill.PRAYER);
+		int IMAGE_SIZE = 16;
+		String counter_health_text;
+		String counter_prayer_text;
+
+		//Store images
+		BufferedImage healthImage = sim.getSkillImage(Skill.HITPOINTS, true);
+		BufferedImage prayerImage = sim.getSkillImage(Skill.PRAYER, true);
+
+		//Code to toggle skill icons
+		if (!config.enableSkillIcon())
+		{
+			healthImage = new BufferedImage(17, 17, BufferedImage.TYPE_INT_ARGB);
+			prayerImage = new BufferedImage(17, 17, BufferedImage.TYPE_INT_ARGB);
+		}
+
+		//Code to toggle counters
+		if (config.enableCounter())
+		{
+			COUNTER_COLOR = new Color (255, 255, 255,  255);
+		}
+		else
+			{
+			COUNTER_COLOR = new Color ( 0, 0, 0,  0);
+		}
+
+		// If the player gets inflicted with poison or venom the colors will be replaced to indicate this status.
+		if (poison_state > 0 && poison_state < 37)
+		{
+			HEALTH_LEFT = POISONED_HEALTH_LEFT;
+		}
+		else if (poison_state > 1000000)
+		{
+			HEALTH_LEFT = VENOMED_HEALTH_LEFT;
+			POISONED_HEALTH_LEFT = VENOMED_HEALTH_LEFT;
+		}
+		else
+		{
+			//reset colors if not poisoned anymore.
+			HEALTH_LEFT = new Color(200, 35, 0);
```

Comment:
Wait what. What is this CONSTANT naming when you are mutating it. make the variables local in method maybe?